### PR TITLE
Documentation: Fix

### PR DESCRIPTION
**Pull request by Alf21**
TODO:
- [x] Add `@realm` to `hook.(Run|Call)`

### CODE REVIEW

File: gamemodes/terrortown/gamemode/server/sv_roleselection.lua [hunk 1]
```diff
@@ -654,8 +671,8 @@ end
 -- Select selectable @{ROLE}s for a given list of @{Player}s
 -- @note This automatically synces with every connected @{Player}
 --
--- @param ?table plys list of @{Player}s. `nil` to calculate automatically
--- @param ?number maxPlys amount of maximum @{Player}s. `nil` to calculate automatically
+-- @param nil|table plys list of @{Player}s. `nil` to calculate automatically
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/server/sv_roleselection.lua:674]: shouldn't it be here `[opt]` as well? (same in the next line)

File: gamemodes/terrortown/gamemode/shared/hud_elements/tttdnascanner/pure_skin_dnascanner.lua [hunk 2]
```diff
@@ -54,6 +56,7 @@ if CLIENT then
 		BaseClass.PerformLayout(self)
 	end
 
+	--- @ignore
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/shared/hud_elements/tttdnascanner/pure_skin_dnascanner.lua:59]: is this ignore needed? This file has an `--- @ignore` in the first line

File: gamemodes/terrortown/gamemode/shared/hud_elements/tttinfopanel/old_ttt_info.lua [hunk 3]
```diff
@@ -11,7 +13,9 @@ DEFINE_BASECLASS(base)
 HUDELEMENT.Base = base
 
 if CLIENT then
-	local hudTeamicon = CreateClientConVar("ttt2_base_hud_teamicon", "1")
+	---
+	-- @realm client
+	local hudTeamicon = CreateConVar("ttt2_base_hud_teamicon", "1", FCVAR_ARCHIVE)
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/shared/hud_elements/tttinfopanel/old_ttt_info.lua:18]: are convars detected if the file is ignored?

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 4]
```diff
@@ -394,6 +437,25 @@ if CLIENT then
...
+	-- @see https://wiki.facepunch.com/gmod/WEAPON:DrawWeaponSelection
+	-- @realm client
+	function SWEP:DrawWeaponSelection()
+
+	end
+
+	---
+	-- @realm client
+	function SWEP:CalcViewModel()
+		if not CLIENT or not IsFirstTimePredicted() then return end
```
Comment (Histalek) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:452]:
```suggestion
		if not IsFirstTimePredicted() then return end
```

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 5]
```diff
@@ -830,115 +919,71 @@ function SWEP:Think()
 	self:CalcViewModel()
```
Comment (Histalek) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:919]:
```suggestion
	if CLIENT then
		self:CalcViewModel()
	end
```

File: gamemodes/terrortown/gamemode/shared/hud_elements/tttinfopanel/old_ttt_info.lua (hunk 3 above)
Comment (Alf21) [reply gamemodes/terrortown/gamemode/shared/hud_elements/tttinfopanel/old_ttt_info.lua:18]: Nope

Comment (TimGoll) [reply gamemodes/terrortown/gamemode/shared/hud_elements/tttinfopanel/old_ttt_info.lua:16]: then you should probably change this

File: lua/bind/cl_init.lua [hunk 6]
```diff
@@ -271,19 +277,24 @@ end
...
 end
 
 ---
 -- Adds an entry to the SettingsBindings table, to easily present them eg. in a GUI.
 -- @param string name
 -- @param string label
--- @param string[opt] category
--- @param number[optchain] defaultKey
+-- @param[opt] string category
+-- @param[optchain] number defaultKey
```
Comment (saibotk) [inline lua/bind/cl_init.lua:296]:
```suggestion
-- @param[opt] number defaultKey
```

**Review by saibotk — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -27,6 +27,8 @@ All notable changes to TTT2 will be documented here. Inspired by [keep a changel
 - Inverted some convars to have a uniform "Enable feature X", not a mixture of enable and disable
 - TargetID text is now scaled with the global scale factor
 - Cleaned up draw function files
+- Changed several functions' scopes
+- Added minimal docu to every datastructure
```
Comment (saibotk) [inline CHANGELOG.md:31]:
```suggestion
- Added minimal documentation to every datastructure
```

File: gamemodes/terrortown/gamemode/shared/hud_elements/base_elements/old_ttt_element.lua [hunk 8]
```diff
@@ -13,8 +13,9 @@ DEFINE_BASECLASS(base)
 HUDELEMENT.Base = base
 
 if CLIENT then
-	local hudWidth = CreateClientConVar("ttt2_base_hud_width", "0")
-	local hudTeamicon = CreateClientConVar("ttt2_base_hud_teamicon", "1")
```
[the remark this answers is not in the document]
Comment (Alf21) [reply gamemodes/terrortown/gamemode/shared/hud_elements/base_elements/old_ttt_element.lua:17]: removed

**Review by saibotk — APPROVED** [review]

**Review by Histalek — APPROVED** [review]
LGTM

**Review by TimGoll — APPROVED** [review]

**Pull request merged**